### PR TITLE
Fix validation of resources (cpu, memory, storage) for limit range types.

### PR DESCRIPTION
**Pull request by aveshagarwal**

### CODE REVIEW

**Comment by aveshagarwal** [comment]
@kubernetes/rh-cluster-infra  @derekwaynecarr

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -2044,6 +2055,9 @@ func ValidateLimitRange(limitRange *api.LimitRange) field.ErrorList {
 	for i := range limitRange.Spec.Limits {
 		idxPath := fldPath.Index(i)
 		limit := &limitRange.Spec.Limits[i]
+		if !validation.IsQualifiedName(string(limit.Type)) {
```
Comment (derekwaynecarr) [inline pkg/api/validation/validation.go]: Can you move this to a separate function and have it check for standard limit types and then require that things that are not standard resource types are qualified (i.e. they have a / ) similar to validateResourceName?

**Comment by derekwaynecarr** [comment]
One request then this is LGTM, thanks @aveshagarwal !

**Comment by aveshagarwal** [comment]
@derekwaynecarr Updated this PR with validation of standard limit types and non standard limit types (should be fully qualified) and also added a test case. PTAL.

File: pkg/api/validation/validation.go [hunk 2]
```diff
@@ -2034,6 +2034,33 @@ func validateResourceQuotaResourceName(value string, fldPath *field.Path) field.
...
 }
 
+// Validate limit range types
+func validateLimitRangeTypeName(value string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if !validation.IsQualifiedName(value) {
+		return append(allErrs, field.Invalid(fldPath, value, qualifiedNameErrorMsg))
+	}
+
+	if len(strings.Split(value, "/")) == 1 {
```
Comment (ncdc) [inline pkg/api/validation/validation.go]: use strings.Contains?

Comment (ncdc) [inline pkg/api/validation/validation.go]: Or do you really want exactly 1 slash?

Comment (aveshagarwal) [inline pkg/api/validation/validation.go]: It seems that in the function IsQualifiedName and also as per this document:
https://github.com/kubernetes/kubernetes/blob/master/docs/design/resources.md 
only one slash is expected to be a qualified name.

Comment (aveshagarwal) [inline pkg/api/validation/validation.go]: With more than one slash, IsQualifiedName fails.

Comment (derekwaynecarr) [inline pkg/api/validation/validation.go]: yes, i am biased to stay consistent, so this lgtm.

**Comment by eparis** [comment]
This PR was sucessfully cherry picked in PR #22855
please verify that the release-1.2 branch contains these changes as you would expect and contact @eparis if there appear to be problems.

**Pull request merged**